fedimint-core/src/admin_client.rs

```diff
@@ -60,15 +73,58 @@ pub struct PeerServerParams {
     /// Name of the peer, used in TLS auth
     pub name: String,
     /// Status of the peer if known
-    pub status: Option<ServerStatus>,
+    pub status: Option<ServerStatusLegacy>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct SetLocalParamsRequest {
+    /// Name of the peer, used in TLS auth
+    pub name: String,
+    /// Federation name set by the leader
+    pub federation_name: Option<String>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq, Encodable, Decodable)]
+/// Connection information sent between peers in order to start config gen
+pub struct PeerConnectionInfo {
```

Comment:
Nit: This seems to contain more than just connectivity info so, maybe we should adjust the name. `PeerSetupInfo` or whatever?